docs/changelog.rst

```diff
@@ -19,12 +19,23 @@ KSQL 5.3.0 includes new features, including:
   This is required to avoid the potential for data loss should this step be dropped.
   See `Github issue #2636 <https://github.com/confluentinc/ksql/pull/2636>`_ for more info.
 
-
 KSQL 5.3.0 includes bug fixes, including:
 
 * The ``ROWTIME`` of the row generated when a ``JOIN`` encounters late data was previous the ``ROWTIME`` of the late event,
   where as now it is the max of ``ROWTIME`` of the rows involved in the join.  This provides more deterministic join semantics.
 
+* Return values of UDF and UDAFs are now correctly marked as optional, where previously there was
+  potential for non-optional fields, which would result in serialization issues in the presence
+  of ``null`` values.
+
+  This is a forward compatible change in Avro, i.e. after upgrading, KSQL will be able to
```

Comment:
From Avro Documentation:
> **if reader's is a union, but writer's is not**
> The first schema in the reader's union that matches the writer's schema is recursively resolved against it. If none match, an error is signalled.
>
> **if writer's is a union, but reader's is not**
> If the reader's schema matches the selected writer's schema, it is recursively resolved against it. If they do not match, an error is signalled.

If we properly have our reader/writer schemas setup then I believe this change is both forwards and backwards compatible for AVRO (forwards because of the first rule and backwards because of the second rule)!